en/dev_setup/dev_env_windows_wsl.md

```diff
@@ -13,70 +25,137 @@ Alternatively QGC for Linux can be run within WSL
+   :::warning
+   This is important!
+   If you work from a location outside of the WSL file system you'll run into issues such as very slow execution and access right/permission errors.
+   :::
+
+1. [Download PX4 Source Code](../dev_setup/building_px4.md):
+   ```bash
+   git clone https://github.com/PX4/PX4-Autopilot.git --recursive
+   ```
+1. Run the **ubuntu.sh** installer script and acknowledge any prompts as the script progresses:
+   ```bash
+   bash ./PX4-Autopilot/Tools/setup/ubuntu.sh
+   ```
+   
+   :::note
+   This installs tools to build PX4 for Pixhawk, Gazebo and JMAVSim targets.
+   - You can use the `--no-nuttx` and `--no-sim-tools` options to omit the NuttX and/or simulation tools.
+   - Other Linux build targets are untested (you can try these by entering the appropriate commands in [Ubuntu Development Environment](../dev_setup/dev_env_linux_ubuntu.md) into the WSL shell.
+     Note that `git` is already installed in WSL.
+   
+1. Restart the "WSL computer" on completion.
+   ```
+   wsl --shutdown
+   ```
+   (or do `exit` and reopen the prompt).
+
 1. Execute command `wsl` to start WSL again and access the shell.
 1. Execute `cd ~/PX4-Autopilot` to switch to the PX4 repository in the WSL home folder.
-1. Execute e.g. `make px4_sitl` to build the PX4 SITL target and test your environment.
+1. Execute `make px4_sitl` to build the PX4 SITL target and test your environment.
+
 
 ### Opening a WSL shell
 
-1. Press the start key.
+To open the WSL shell (once WSL has been installed):
+
+1. Press the Windows **Start** key.
 1. Type "cmd" and press Enter to open a command prompt.
 1. Execute the command `wsl` to access the WSL shell.
 
+
 ## Visual Studio Code Integration
 
 1. Download and install Visual Studio Code on Windows from https://code.visualstudio.com/
-2. Open Visual Studio Code
-3. Install the extension called "Remote - WSL": https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl
+2. Open _Visual Studio Code_.
+3. Install the extension called [Remote - WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) (marketplace) 
 4. [Open a WSL shell](dev_env_windows_wsl.md#opening-a-wsl-shell)
-5. Switch to the PX4 folder `cd ~/PX4-Autopilot`
+5. Switch to the PX4 folder: `cd ~/PX4-Autopilot`
 6. Open this folder with WSL integration in Visual Studio Code `code .`
-7. Now you have the full IDE integration with Terminal, make sure you always open the PX4 repository in the Remote WSL mode. It will be available in your recently opened locations whenever you start VS Code.
+7. Now you have the full IDE integration with Terminal, make sure you always open the PX4 repository in the Remote WSL mode.
+   It will be available in your recently opened locations whenever you start VS Code.
+
+## QGroundControl
+
+You can run QGroundControl in either WSL or Windows to connect to the running simulation.
```

Comment:
You can either run the Linux version of QGC within WSL or the Windows version of QGC outside of WSL.
If QGC is ran inside WSL it will automatically connect to your simulation but show through the WSLg graphics layer and is not be able to flash a flight control board. If you run QGC for Windows it runs native and can flash flight control boards but you have to manually set up the communication link to the simulation running in WSL.

Does that make sense?